Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -151,39 +293,39 @@ void CookieStore::getAll(CookieStoreGetOptions&& options, Ref<DeferredPromise>&&
         return;
     }
 
-    if (!m_cookieJar) {
-        promise->reject(SecurityError);
-        return;
-    }
-
-    auto& document = *downcast<Document>(context);
-    auto url = document.url();
-
+    auto url = context->url();
     if (!options.url.isNull()) {
-        auto parsed = document.completeURL(options.url);
-        if (scriptExecutionContext()->isDocument() && parsed != url) {
+        auto parsed = context->completeURL(options.url);
+        if (context->isDocument() && parsed != url) {
             promise->reject(TypeError);
             return;
         }
+
         if (!origin->isSameOriginDomain(SecurityOrigin::create(parsed))) {
             promise->reject(TypeError);
             return;
         }
         url = WTFMove(parsed);
     }
 
-    auto completionHandler = [promise = WTFMove(promise)] (std::optional<Vector<Cookie>>&& cookies) {
-        if (!cookies) {
-            promise->reject(TypeError);
+    m_promises.add(++m_nextPromiseIdentifier, WTFMove(promise));
+    auto completionHandler = [promiseIdentifier = m_nextPromiseIdentifier](CookieStore& cookieStore, ExceptionOr<Vector<Cookie>>&& result) {
+        auto promise = cookieStore.takePromise(promiseIdentifier);
+        if (!promise)
+            return;
+
+        if (result.hasException()) {
+            promise->reject(result.releaseException());
             return;
         }
 
-        promise->resolve<IDLSequence<IDLDictionary<CookieListItem>>>(WTF::map(WTFMove(*cookies), [](auto&& cookie) {
+        auto cookiesVector = result.releaseReturnValue();
```

Comment:
Same comment here.

Reply:
If I have the cookies vector as a reference then I won't able to pass `WTFMove(cookies)` to the `map` function below.

Reply:
Keeping releaseReturnValue for now.